index.js

```diff
@@ -0,0 +1,64 @@
+const express = require("express");
+const app = express();
+const axios = require("axios");
+const pool = require("./database");
+require("dotenv").config();
+
+app.use(express.json());
+
+let port = process.env.PORT || 3000;
+
+async function get_github_discord_url_by_id() {
+  const data_1 = await pool.query(
+    `select discord_url from github_discord_url where id = $1`,
+    [`${db_id}`]
+  );
+
+  if (data_1.rows === null || data_1.rows >= 2 || data_1.rows.length == 0) {
+    throw new Error("row doesn't exist");
+  }
+
+  const github_discord_url = await data_1.rows[0];
+
+  return github_discord_url["discord_url"];
+}
+app.listen(port, () => {
+  console.log(`running on ${port}`);
+});
+
+app.post("/github_push_webhook/:id", async (req, res) => {
+  try {
+    const { id } = req.params;
+    var reqData = req.body;
+
+    if (reqData.head_commit === undefined) {
+      return res.status(200).json({ message: "Github is testing" });
+    }
+
+    if (
+      reqData.repository.owner.login === null ||
+      reqData.repository.name === null
+    ) {
+      return res.status(400).json({ error: "bad format" });
+    }
+
+    get_github_discord_url_by_id((db_id = id));
```

Comment:
why is this function called twice?

Reply:
Because once it is sending data to the function and second time it is getting data from the function

Reply:
this doesn't make sense; you should be calling it only once with the params.